test/acme/dns/fixture.go

```diff
@@ -72,78 +70,37 @@ type fixture struct {
 	// Default: "123d=="
 	dnsChallengeKey string
 
-	// controlPlane is a reference to the control plane that is used to run the
-	// test suite.
-	// It is constructed when a Run* method is called.
-	controlPlane *integration.ControlPlane
-	restConfig   *rest.Config
-	clientset    kubernetes.Interface
-	kubectl      *integration.KubeCtl
-	setupLock    sync.Mutex
+	setupLock   sync.Mutex
+	environment *envtest.Environment
+	clientset   kubernetes.Interface
 
 	pollInterval     time.Duration
 	propagationLimit time.Duration
 }
 
-var DefaultKubeAPIServerFlags = []string{
-	"--etcd-servers={{ if .EtcdURL }}{{ .EtcdURL.String }}{{ end }}",
-	"--cert-dir={{ .CertDir }}",
-	"--insecure-port={{ if .URL }}{{ .URL.Port }}{{ end }}",
-	"--insecure-bind-address={{ if .URL }}{{ .URL.Hostname }}{{ end }}",
-	"--secure-port={{ if .SecurePort }}{{ .SecurePort }}{{ end }}",
-	"--admission-control=AlwaysAdmit",
-}
-
-// Setup will set up the test fixture by running kube-apiserver and etcd.
-// One instance of the apiserver and etcd will be shared throughout all of the
-// suite.
-// The first time this function is called, the function that is returned will
-// be the control plane's Stop function. Subsequent calls to setup will return
-// a function that does nothing. This allows all the Run* functions to call
-// setup, and defer cleaning up the fixture, but only the first 'entrypoint'
-// Run function will actually clean up the apiserver.
-func (f *fixture) setup(t *testing.T) func() error {
+func (f *fixture) setup(t *testing.T) func() {
 	f.setupLock.Lock()
 	defer f.setupLock.Unlock()
 
 	if err := validate(f); err != nil {
 		t.Fatalf("error validating test fixture configuration: %v", err)
 	}
 
-	if f.controlPlane != nil {
-		return func() error { return nil }
-	}
-	f.controlPlane = &integration.ControlPlane{}
-	f.controlPlane.APIServer = &integration.APIServer{
-		Args: DefaultKubeAPIServerFlags,
-		Path: f.binariesPath + "/kube-apiserver",
-	}
-	f.controlPlane.Etcd = &integration.Etcd{
-		Path: f.binariesPath + "/etcd",
-	}
-	if err := f.controlPlane.Start(); err != nil {
-		t.Fatalf("error starting apiserver: %v", err)
-	}
-	t.Logf("started apiserver on %q", f.controlPlane.APIURL())
-	// Create the *rest.Config for creating new clients
-	f.restConfig = &rest.Config{
-		Host: f.controlPlane.APIURL().Host,
-		// gotta go fast during tests -- we don't really care about overwhelming our test API server
-		QPS:   1000.0,
```

Comment:
Question: This custom QPS/Burst config got removed. Should we keep that?

Reply:
in `envtest.Environment`, this logic is also present:
https://github.com/kubernetes-sigs/controller-runtime/blob/985e819db7af1cd17782ca7ad27644d0ca655054/pkg/envtest/server.go#L265-L269